diff --git a/Makefile b/Makefile
index 263a6acfd6..fe9569d200 100644
--- a/Makefile
+++ b/Makefile
@@ -1855,7 +1855,7 @@ $(filter-out tools, $(u-boot-dirs)): tools
 examples: $(filter-out examples, $(u-boot-dirs))
 
 define filechk_uboot.release
-	echo "$(UBOOTVERSION)$$($(CONFIG_SHELL) $(srctree)/scripts/setlocalversion $(srctree))"
+	echo "$(UBOOTVERSION)$$($(CONFIG_SHELL) $(srctree)/scripts/setlocalversion $(srctree))-imx-6.6.3-1.0.0-adv"
 endef
 
 # Store (new) UBOOTRELEASE string in include/config/uboot.release
