Observation: ad hoc patch
Upstream-Status: Inappropriate [other]
Signed-off-by: F. Aragon <francisco.aragon@advantech.de>

diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index f164e353c..b31fa9c5f 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -429,7 +429,7 @@ CFG_UART_BASE ?= UART1_BASE
 endif
 
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mmevk))
-CFG_DDR_SIZE ?= 0x80000000
+CFG_DDR_SIZE ?= 0x40000000
 CFG_UART_BASE ?= UART2_BASE
 endif
 
